internal/consumer/dlq.go

```diff
@@ -181,3 +194,47 @@ func (d *bufferedErrorTopic) newSaramaMessage(key, value []byte, responseC chan
 		Metadata: responseC,
 	}
 }
+
+// NewRetryDLQMultiplexer returns a DLQ that will produce messages to retryTopic or dlqTopic depending on
+// the threshold.
+//
+// Messages that are added to this DLQ will be sent to retryTopic if the retry count of the message is
+// < the threshold.
+// Else, it will go to the dlqTopic.
+func NewRetryDLQMultiplexer(retryTopic, dlqTopic DLQ, threshold int64) DLQ {
+	return &retryDLQMultiplexer{
+		retryCountThreshold: threshold,
+		retryTopic:          retryTopic,
+		dlqTopic:            dlqTopic,
+	}
+}
+
+// Add sends a kafka message to the retry topic or dlq topic depending on the retry count in the message.
+//
+// If the message RetryCount is greater than or equal to the retryCountTreshold in the multiplexer,
+// the message will be sent to the retry topic.
+// Else, it will be sent to the dlq topic.
+func (d *retryDLQMultiplexer) Add(m kafka.Message) error {
+	if m.RetryCount() >= d.retryCountThreshold {
+		return d.dlqTopic.Add(m)
+	}
```

Comment:
I believe you need to increment the retryCount here. Can you also add a test for this ?

Reply:
I don't think you need to increment before the check (it will be incremented in the Metadata writer later. Note that I used >= there.